### PR TITLE
[WIP] Download path can be changed in the settings

### PR DESCRIPTION
**Pull request by jetownfeve21**
need https://github.com/kiwix/kiwix-lib/pull/310
https://github.com/kiwix/kiwix-desktop/issues/170

### CODE REVIEW

**Comment by jetownfeve21** [comment]
@mgautierfr Can I have your review about this before starting to work on the profile path ?

**Comment by kelson42** [comment]
@mgautierfr Please review this PR.

File: src/contentmanager.cpp [hunk 1]
```diff
@@ -214,12 +217,13 @@ QString ContentManager::downloadBook(const QString &id)
     return QString::fromStdString(download->getDid());
 }
 
-void ContentManager::eraseBookFilesFromComputer(const QString fileToRemove)
+void ContentManager::eraseBookFilesFromComputer(const QString bookPath)
```
Comment (mgautierfr) [inline src/contentmanager.cpp]: I think this method is a bit too complex.
We only use this method in eraseBook and cancelBook. There, we create a Qstring from the book path and then we split it in the function to get the different parts.

We should change the signature of the method to directly give the different information we need.

File: src/settingsmanager.cpp [hunk 2]
```diff
@@ -61,8 +65,35 @@ void SettingsManager::setZoomFactor(qreal zoomFactor)
...
+                                                    QString(),
+                                                    QFileDialog::ShowDirsOnly);
+    setDownloadPath(dir);
+}
+
 void SettingsManager::initSettings()
 {
     m_kiwixServerPort = m_settings.value("localKiwixServer/port", 8181).toInt();
     m_zoomFactor = m_settings.value("view/zoomFactor", 1).toDouble();
+    m_downloadPath = m_settings.value("download/path", QString::fromStdString(getDataDirectory())).toString();
```
Comment (mgautierfr) [inline src/settingsmanager.cpp]: I would prefer "download/dir" (and `downloadDir` in all the method/variable name).

The `dir` is explicitly a directory. Naively, I would said that the download path is the full path of the zim file, not the directory where we are downloading it.

**Comment by kelson42** [comment]
@jetownfeve21 
* Please remove "WIP" from the PR if your PR is ready
* Please put properly "Fixes #XYZ" to close the ticket automatically. If this is should not closing the ticket, please make a comment explaining what this PR does.

**Pull request merged**